src/TypeParser.php

```diff
@@ -54,4 +54,39 @@ public function boolean()
     {
         return new BooleanParser($this->exceptionFactory, $this->name, $this->value);
     }
+
+    public function commaSeparatedString()
```

Comment:
IMO syntax should be `$this->queryParameter(..)->commaSeparated()->string()` instead of `$this->query(..)->commaSeparatedString()`. Without this, the autocompletion is full of commaSeperated stuff, but the commaSeparated stuff is not used very often.

`TypeParser::commaSeparated` just needs to return it's own `CommaSeparatedTypeParser` object :)